infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -235,9 +254,13 @@ public synchronized void tokenRequest(CorfuPayloadMsg<TokenRequest> msg,
                                           ChannelHandlerContext ctx, IServerRouter r) {
         TokenRequest req = msg.getPayload();
 
+        boolean isEnabled = MetricsUtils.isMetricsCollectionEnabled();
```

Comment:
once the above is implemented, we should have a per-server metric-handler(msg ...) , which should handle this.